pennylane_lightning/core/src/simulators/lightning_kokkos/StateVectorKokkos.hpp

```diff
@@ -680,6 +705,31 @@ class StateVectorKokkos final
         }
     }
 
+    /**
+     * @brief Apply a GlobalPhase operator to the state vector using a matrix
+     *
+     * @param wires Wires to apply gate to.
+     * @param inverse Indicates whether to use adjoint of gate.
+     * @param params parameters for this gate
+     */
+    void applyGlobalPhase(const std::vector<size_t> &wires,
+                          bool inverse = false,
+                          const std::vector<fp_t> &params = {}) {
+        auto &&num_qubits = this->getNumQubits();
+
+        if (!inverse) {
+            Kokkos::parallel_for(
+                Kokkos::RangePolicy<KokkosExecSpace>(0, exp2(num_qubits)),
+                globalPhaseFunctor<fp_t, false>(*data_, num_qubits, wires,
+                                                params));
+        } else {
+            Kokkos::parallel_for(
+                Kokkos::RangePolicy<KokkosExecSpace>(0, exp2(num_qubits)),
+                globalPhaseFunctor<fp_t, true>(*data_, num_qubits, wires,
+                                               params));
+        }
+    }
```

Comment:
```suggestion
    void applyGlobalPhase(const std::vector<size_t> &wires,
                          const bool inverse = false,
                          const std::vector<fp_t> &params = {}) {
        auto &&num_qubits = this->getNumQubits();

       Kokkos::parallel_for(
                Kokkos::RangePolicy<KokkosExecSpace>(0, exp2(num_qubits)),
                globalPhaseFunctor<fp_t, inverse>(*data_, num_qubits, wires,
                                               params));
    }
```

Reply:
Same, we can't use a variable (even `const`) in a template parameter.

Reply:
Are you sure? Because I tested it with a std::array and it worked. :laughing:

Reply:
Anyway, you don't need to spend time on that, we can leave it like this for now.

Reply:
This does not work on my side, but you're right std::array templates are somehow more flexible.